src/main/java/com/taboola/async_profiler/api/AsyncProfilerService.java

```diff
@@ -0,0 +1,127 @@
+        }
+    }
+
+    public ProfileResult profile(ProfileRequest profileRequest) {
+        return asyncProfilerFacade.profile(profileRequest);
+    }
+
+    public ProfileResult stop() {
+        return asyncProfilerFacade.stop();
+    }
+
+    public void startContinuousProfiling(ProfileRequest profileSnapshotRequest) {
+        synchronized (lock) {
+            if (currentContinuousProfilingTask == null) {
+                RecurringRunnable continuousProfilingTask = new RecurringRunnable(() -> profileAndReportSnapshot(profileSnapshotRequest));
+                continuousProfilingExecutorService.submit(continuousProfilingTask);
+                currentContinuousProfilingTask = continuousProfilingTask;
+            } else {
+                throw new IllegalStateException("Failed starting continuous profiling, there is already an active session");
+            }
+        }
+    }
+
+    public void stopContinuousProfiling() {
+        synchronized (lock) {
+            if (currentContinuousProfilingTask != null) {
+                currentContinuousProfilingTask.cancel();
+                currentContinuousProfilingTask = null;
+                safeStop();
+            } else {
+                throw new IllegalStateException("There is no active continuous profiling session");
+            }
+        }
+    }
+
+    public String getSupportedEvents() {
+        return asyncProfilerFacade.getSupportedEvents();
+    }
+
+    public String getProfilerVersion() {
+        return asyncProfilerFacade.getProfilerVersion();
+    }
+
+    @SneakyThrows
+    private void profileAndReportSnapshot(ProfileRequest profileSnapshotRequest) {
+        try {
+            ProfileResult profileSnapshotResult = asyncProfilerFacade.profile(profileSnapshotRequest);
+            reportAsync(profileSnapshotResult);
+
+        } catch (RuntimeException ex) {
+            //sleep for backoff seconds if the profiler has failed or if we failed to submit the result to the reporter executor service
+            //we will throw from here only if we failed to sleep because it was interrupted, which is ok
+            threadUtils.sleep(continuousProfilingFailureBackoffSeconds, TimeUnit.SECONDS);
+        }
+    }
+
+    private void reportAsync(ProfileResult profileSnapshotResult) {
+        try {
+            snapshotsReporterExecutorService.submit(() -> reportSnapshot(profileSnapshotResult));
+        } catch (Exception ex) {
+            if (profileSnapshotResult != null) {
+                try {
+                    profileSnapshotResult.close();
+                } catch (Exception e) {
+                }
+            }
+
+            throw new RuntimeException(ex);
+        }
+    }
+
+    private void reportSnapshot(ProfileResult snapshotResult) {
+        try (ProfileResult profileResult = snapshotResult) {
+            snapshotsReporter.report(profileResult);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
```

Comment:
Yes, same thought, just wondered where/how to do that exactly (again, didn't want extra dependencies).
Maybe metrics? i'll expose an interface to to some metrics factory that can be provided to create counters etc?

Reply:
You think addin slf4j as dependency is too much? anyway metrics is cool but not explanatory as to why it doesn't work.
If you're really against slf4j, what about keeping the logs in some list and exposing them as endpoint? it's ugly but atleast it will help people debug in case of issues in the system

Reply:
no, you're right, added slf4j